java/src/main/java/com/vmware/App.java

```diff
@@ -0,0 +1,88 @@
+package com.vmware;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Sumit Deo (deosu@vmware.com)
+ */
+public class App {
+
+    private final static Logger logger = LoggerFactory.getLogger(App.class);
+    public static final String OTEL_RESOURCE_ATTRIBUTES_KEY = "otel.resource.attributes";
+    public static final String OTEL_RESOURCE_ATTRIBUTES_VALUE = "service.name=HelloWorldJavaService";
+    public static final String INSTRUMENTATION_LIBRARY_NAME = "instrumentation-library-name";
+    public static final String INSTRUMENTATION_VERSION = "1.0.0";
+    static Tracer tracer;
+
+    public static void main(String[] args) throws InterruptedException {
+
+        /*this will make sure that a proper service.name attribute is set on all the
+         spans/metrics.*/
+        System.setProperty(OTEL_RESOURCE_ATTRIBUTES_KEY, OTEL_RESOURCE_ATTRIBUTES_VALUE);
+
+        /*tracer must be acquired, which is responsible for creating spans and interacting with the Context*/
+        tracer = getTracer();
+
+        /*an automated way to propagate the parent span on the current thread*/
+        for (int index = 0; index < 3; index++) {
+            /*create a span by specifying the name of the span. The start and end time of the span is automatically set by the OpenTelemetry SDK*/
+            Span parentSpan = tracer.spanBuilder("parentSpan").startSpan();
+            logger.info("In parent method. TraceID : {}", parentSpan.getSpanContext().getTraceIdAsHexString());
+
+            /*put the span into the current Context*/
+            try (Scope scope = parentSpan.makeCurrent()) {
+
+                /*annotate the span with attributes specific to the represented operation, to provide additional context*/
+                parentSpan.setAttribute("parentIndex", index);
+                childMethod(parentSpan);
+            } catch (Throwable throwable) {
+                parentSpan.setStatus(StatusCode.ERROR, "Something wrong with the parent span");
```

Comment:
Might be nice to include the throwable's message as a span log?

Reply:
added throwable message in the exception message of a parentSpan.